wolox-books/src/app/screens/login/login.component.ts

```diff
@@ -7,9 +12,39 @@ import { Component, OnInit } from '@angular/core';
 })
 export class LoginComponent implements OnInit {
 
-  constructor() { }
+  form: FormGroup;
+  public Patterns = patterns;
+
+  constructor(
+    private formBuilder: FormBuilder,
+    private authService: AuthService,
+    private router: Router
+  ) {
+    this.buildForm();
+   }
 
   ngOnInit() {
   }
 
+  private buildForm() {
```

Comment:
No need for the `private` 🤔

Reply:
It's just because only this file will use this function. If perhaps the HTML will use this, it must  be public.